frontend/src/data/variables/VaccineProvider.ts

```diff
@@ -67,104 +63,14 @@ class VaccineProvider extends VariableProvider {
-        consumedDatasetIds = consumedDatasetIds.concat(
-          "decia_2010_territory_population-by_race_and_ethnicity_territory_state_level"
-        );
-
-        const acs = new DataFrame(acsQueryResponse.data);
-        df = joinOnCols(df, acs, ["fips", breakdownColumnName], "left");
-      }
-
-      df = df.renameSeries({
-        population_pct: "vaccine_population_pct",
-      });
-
-      df = df
-        .generateSeries({
-          vaccinated_pct_share: (row) => row["vaccinated_share_of_known"],
-        })
-        .resetIndex();
+      consumedDatasetIds = consumedDatasetIds.concat(
+        GetAcsDatasetId(breakdowns)
+      );
     } else if (breakdowns.geography === "state") {
-      df = df
-        .generateSeries({
-          vaccinated_pct_share: (row) =>
-            row.vaccinated_pct_share == null ||
-            isNaN(row.vaccinated_pct_share) ||
-            row.vaccinated_pct_share === 0
-              ? null
-              : Math.round(row.vaccinated_pct_share * 100),
-        })
-        .resetIndex();
-
-      df = df
-        .generateSeries({
-          population_pct: (row) =>
-            isNaN(row.population_pct) ||
-            row.population_pct == null ||
-            row.population_pct === 0
-              ? null
-              : Math.round(row.population_pct * 100),
-        })
-        .resetIndex();
-
-      df = df
-        .generateSeries({
-          vaccinated_per_100k: (row) =>
-            isNaN(row.vaccinated_pct) ||
-            row.vaccinated_pct == null ||
-            row.vaccinated_pct === 0
-              ? null
-              : Math.round(row.vaccinated_pct * 1000 * 100),
-        })
-        .resetIndex();
-
-      df = df
-        .renameSeries({
-          population_pct: "vaccine_population_pct",
-        })
-        .resetIndex();
-
-      const acsDatasetId = GetAcsDatasetId(breakdowns);
-      consumedDatasetIds = consumedDatasetIds.concat(acsDatasetId);
-
-      // We have to separate there because the ALL rows contain raw numbers
-      // while the other rows are pre computed
-      let totalDf = df.where((row) => row[breakdownColumnName] === ALL);
-      const nonTotalDf = df.where((row) => row[breakdownColumnName] !== ALL);
-
-      totalDf = totalDf
-        .generateSeries({
-          vaccinated_per_100k: (row) =>
-            this.calculations.per100k(
-              row.vaccinated_first_dose,
-              row.population
-            ),
-        })
-        .resetIndex();
-
-      totalDf = totalDf
-        .generateSeries({
-          vaccinated_pct_share: (row) => 100,
-        })
-        .resetIndex();
-
-      df = totalDf.concat(nonTotalDf).resetIndex();
-
-      df = df
-        .generateSeries({
-          vaccinated_share_of_known: (row) => row["vaccinated_pct_share"],
-        })
-        .resetIndex();
+      consumedDatasetIds = consumedDatasetIds.concat(
+        "acs_population-by_race_state",
+        "acs_2010_population-by_race_and_ethnicity_territory"
```

Comment:
this should be `decia_2020_territory_population-by_race_and_ethnicity_territory_state_level` now right? And there is some more conditional logic, like the territory pops are only used on national reports with undefined `filterFips`, or on state level reports where `isIslandArea(filterFips) =true`